app/DTO/DiagnosticInfo.php

```diff
@@ -2,45 +2,20 @@
 
 namespace App\DTO;
 
-class DiagnosticInfo extends DTO
+class DiagnosticInfo extends ArrayableDTO
```

Comment:
I like the approach to use a common parent class which uses reflection to serialize the properties into an array. I also understand (somehow) the benefits of the benefit of declaring DTO properties via constructor promotion.

But maybe we should pull out this change into an independent PR. This independent PR could collect all changes due to this new approach which are not related to changes due to rights and capabilities. This independent PR could also host the discussion about constructor promotion in the mixed case which I initiated some comments above.

Moreover, I believe that this independent PR is something which we can decide faster than this whole rights and capability stuff.